### ISSUE TITLE
Error use leading app bar

### ISSUE THREAD

**Issue by mazidahmad** [peiffer-innovations/json_dynamic_widget#30]
Hello. First, i have to say thank you to contributors for develop this package. Its great.
Let me show my issue to build an app bar with leading inside that.  

![image](https://user-images.githubusercontent.com/45102623/132283415-da328471-6805-4a7d-9b37-2295de4d6c49.png)

Is that correct to create leading in app bar? coz i can't found sample code with leading from the example project. I just think that leading is a widget type, so i put the object to leading map. But, i got this error

![image](https://user-images.githubusercontent.com/45102623/132283812-12ee4bc5-9dc3-4a64-b003-41cef85370a4.png)

I have read the schema, but the leading are support type of string, number, and boolean. Even though the leading is a widget object in material.

Thanks in advance

**Comment by jpeiffer** [comment]
Can you post a full example of a JSON file with the issue you're having?  It doesn't have to be the exact one you're trying, just any JSON showing the issue.  With that, I can try to help.

**Comment by jpeiffer** [comment]
Does it work for you in `profile` or `release` mode?  I actually think I might know the issue.  If it works in one of those that confirms my theory.

**Comment by mazidahmad** [comment]
I'm using a lot of custom widgets in my project. Maybe its example from a simple JSON widget from my issue.

```
{
  "type": "scaffold",
  "args": {
    "appBar": {
      "type": "app_bar",
      "args": {
        "backgroundColor": "#000000",
        "title": {
          "type": "text",
          "args": {
            "text": "Leading Example"
          }
        },
        "elevation": 0,
        "automaticallyImplyLeading": false,
        "leading": {
          "type": "icon_button",
          "args": {
            "icon": {
              "type": "icon",
              "args": {
                "icon": {
                  "codePoint": 58712,
                  "fontFamily": "MaterialIcons",
                  "color": "#FFFFFF"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

I think it should build the widget, cause I see the code in the library especially in app_bar_builder. And I found the leading param and should to build by another widget.

![image](https://user-images.githubusercontent.com/45102623/134614642-9d772304-5c05-4cfb-9f5d-e412b15ff3b1.png)

And the issue when I want to use automaticallyImplyLeading with false value, but it throws to exception. 

I can't debug the code of the library coz it from another package. 

Thanks a lot.

**Comment by jpeiffer** [comment]
Thanks a lot for both the report and the JSON.  I published v2.2.0 today which includes your example in working form now.

I'm going to close this issue as I believe it is resolved.  Please feel free to re-open it if you disagree and believe it still needs more work.

**Comment by mazidahmad** [comment]
> Thanks a lot for both the report and the JSON. I published v2.2.0 today which includes your example in working form now.
> 
> I'm going to close this issue as I believe it is resolved. Please feel free to re-open it if you disagree and believe it still needs more work.

I've tried it with v2.2.0 and now it works like a charm. Thanks in advance.
